common/sketch/src/main/java/org/apache/spark/util/sketch/Murmur3_x86_32.java

```diff
@@ -92,8 +96,10 @@ private static int hashBytesByInt(Object base, long offset, int lengthInBytes, i
     int h1 = seed;
     for (int i = 0; i < lengthInBytes; i += 4) {
       int halfWord = Platform.getInt(base, offset + i);
-      int k1 = mixK1(halfWord);
-      h1 = mixH1(h1, k1);
+      if (isBigEndian) {
```

Comment:
I am not sure how much difference it makes, but this condition is of course checked every time in the loop.
If one were willing to duplicate the loop, you could check it once at the outset. But unless there's evidence that is non-trivial overhead and that this is a very hot loop, probably the extra branch is negligible, esp. if it is smart enough to know it's always false/true.

Reply:
cc @rednaxelafx

Reply:
also @kiszk

Reply:
I agree to move this condition out of the loop while the overhead of the condition itself may be small.   
I expect that the HotSpot compiler would/could generate one native instruction `reverse load` with the adjacent getInt and reverseBytes.

Reply:
I'm inclined to agree with @srowen that duplicating the loop is probably unnecessary. Hotspot will almost certainly be able to remove the branch and even if it doesn't this code isn't actually in the critical path for the hash function (the critical path is through mixH1) so the latency of the extra check can be hidden on an out of order processor. I'm happy to do it if the consensus is that it is warranted though.